sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/Validations.scala

```diff
@@ -121,8 +126,13 @@ object Validations {
       commonValidation(component) ++
       commonSubscriptionValidation(component) ++
       viewMustHaveTableName(component) ++
-      viewMustHaveOneQueryMethod(component) ++
       streamUpdatesQueryMustReturnFlux(component)
+    } ++
+    whenExcluding[View[_], ViewTable[_]](component) {
```

Comment:
I didn't find so clear which is the typed that needs to match and which is the type that must not match. Maybe an if and else is better, for instance...

```scala
    val commonViewValidations =
      commonValidation(component) ++
      commonSubscriptionValidation(component) ++
      viewMustHaveTableName(component) ++
      streamUpdatesQueryMustReturnFlux(component)

    if (assignable[ViewTable[_]](component)) {
      commonViewValidations
    } else if (assignable[View[_]](component)) {
      commonViewValidations ++ viewMustHaveOneQueryMethod(component)
    } else if (assignable[MultiTableView](component)) {
      viewMustHaveOneQueryMethod(component)
    } else {
      Valid
    }
```

Reply:
Or maybe this...

```scala
private def whenNot[T: ClassTag](component: Class[_])(block: => Validation): Validation = 
  if (!assignable[T](component)) block else Valid
```

```scala
    when[View[_]](component) {
      commonValidation(component) ++
      commonSubscriptionValidation(component) ++
      viewMustHaveTableName(component) ++
      streamUpdatesQueryMustReturnFlux(component) ++
      whenNot[ViewTable[_]](component) {
        viewMustHaveOneQueryMethod(component)
      }
    } ++
    when[MultiTableView](component) {
      viewMustHaveOneQueryMethod(component)
    }
```

Reply:
And if make `component` an implicit? 😄 

```scala
when[View[_]] {
  whenNot[ViewTable[_]] {
    viewMustHaveOneQueryMethod
  } ++
  commonValidation ++
  commonSubscriptionValidation ++
  viewMustHaveTableName ++
  streamUpdatesQueryMustReturnFlux
} ++
when[MultiTableView] {
  viewMustHaveOneQueryMethod
}
```

Reply:
With the `ViewTable` and `MultiTableView` classes removed, this needed to be different anyway.